WolvenKit.App/ViewModels/Documents/RDTDataViewModel.cs

```diff
@@ -87,6 +91,65 @@ public List<ChunkViewModel> Chunks
 
         #region commands
 
+        public ICommand OnDemandLoadingCommand { get; private set; }
+
+        private bool CanExecuteOnDemandLoading(TreeViewNode node)
+        {
+            if (node.Content is GroupedChunkViewModel)
+            {
+                return true;
+            }
+
+            if (node.Content is ChunkViewModel cvm && cvm.HasChildren())
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        private void ExecuteOnDemandLoading(TreeViewNode node)
+        {
```

Comment:
Maybe this could be encapsulated behind a gcvm/cvm `DoExpand()` or a notif so that we don’t need to know the internals plus the node can decide whether it is or isn’t expanded?